src/Avalonia.X11/X11Window.cs

```diff
@@ -1449,7 +1441,7 @@ public void SetFrameThemeVariant(PlatformThemeVariant themeVariant) { }
 
         public bool NeedsManagedDecorations => false;
 
-        public bool IsEnabled => !_disabled;
+        public bool IsEnabled => !_disabled && !_mode.BlockInput;
```

Comment:
"ModalityOn" doesn't necessary mean "Window is disabled", or does it in X11? We also don't notify anybody, when BlockInput is changed. Is it expected?

Reply:
https://specifications.freedesktop.org/xembed-spec/latest/messages.html#id-1.6.16
>An embedded control should ignore mouse input while modality is active. Note that that keyboard input is blocked anyway by XEmbed, since the topmost embedder will not pass keyboard events through in modal state.